### PR TITLE
Kirby adjustments

### PR DESCRIPTION
**Pull request by SkewedAskew**
Big changes for a small puffball

Assets: [assets-kirb.zip](https://github.com/HDR-Development/HewDraw-Remix/files/11683859/assets-kirb.zip)

Hitbox comparisons: https://imgur.com/a/xB03Kc5

### Jab 1:
- [-] Removed pull-in hitbox
- [+] Hitboxes moved forward
- [+] Hitbox Duration: F2-3 -> F2-4
- [/] Hitlag Multiplier: 1.7x -> 1.0x
- [+] Hitbox Size: 2.5u -> 3.0u
- [/] Jab Transition Frame: F5 -> F6

### Jab 2:
- [+] Hitbox Duration: F3-4 -> F3-5
- [/] Hitlag Multiplier: 1.2x -> 1.0x
- [-] SDI Multiplier: 0.3x -> 1.0x
- [/] Jab Transition Frame: F5 -> F6

### Forward Tilt:
- High
  - [+] Damage: 8.0/7.0% -> 9.0%
  - [R] Angle: 361 -> 60
  - [R] BKB: 40 -> 60
  - [R] KBG: 100 -> 45
- Mid
  - [+] Damage: 8.0/7.0% -> 9.0%
- Low
  - [+] Damage: 8.0/7.0% -> 9.0%
  - [R] Angle: 361 -> 35
  - [R] BKB: 40 -> 60
  - [R] KBG: 100 -> 35

### Up Tilt:
- Early
  - [+] Angle: 100/98 -> 98
  - [/] BKB: 45 -> 55
  - [/] KBG: 105/101 -> 80
- Late
  - [+] Angle: 88/86 -> 88
  - [/] BKB: 45 -> 55
  - [/] 105/101 -> 80

### Down Tilt:
- [/] BKB: 40 -> 35
- [/] KBG: 40 -> 50

### Dash Attack:
- (!) Momentum reworked; travels faster at the start and slower at the end
- (!) Can now slide off edges between frames 1-21
- [R] Removed mid hit
- [-] Hitbox Duration (early/late): F9-13/F14-19/F20-30 -> F9-12/F13-24
- Early
  - [+] Damage: 12.0% -> 13.0%
  - [/] Hitlag Multiplier: 1.3x -> 1.0x
- Late
  - [+] Damage: 6.0% -> 9.0%
  - [-] Angle: 86 -> 70
  - [/] Hitlag Multiplier: 0.8x -> 1.0x
  - [+] Hitbox Size: 3.5u -> 4.0u
- [+] FAF: 50 -> 47

### Up Smash
- [+] Hitbox Duration (early/mid/late): F12-14/F15-16/F17 -> F11-13/F14-15/F16
- [+] FAF: 48 -> 47

### Down Smash:
- [+] Added inner foot and body hitboxes
- [+] Hitbox Duration (early/late): F10-14/F15-23 -> F7-11/F12-18
- Body
  - Damage: 10.0%
  - Angle: 90
  - BKB: 80
  - KBG: 40
  - Hitbox Size: 4.0u
- Outer Foot
  - [-] Hitbox Size: 4.0u -> 3.5u
- [+] FAF: 60 -> 45

### Forward Air:
- Hit 1/Hit 2
  - [-] Additional Hitstun: 1F -> 0F

### Back Air:
- [-] Hitbox Duration (early/late): F6-8/F9-22 -> F6-9/F10-16

### Up Air
- [R] BKB: 20 -> 25
- [R] KBG: 115 -> 90

### Down Air:
- [R] Removed final hit
- [+] Changed from two-frame hitbox windows to a constant rehit rate
- Ground
  - [+] Damage: 1.0% -> 2.0%
  - [-] BKB: 47 -> 40
  - [-] KBG: 40 -> 36
  - [+] SDI Multiplier: 1.2x -> 1.0x
  - (!) Frame Advantage (0/100%): +9/+14 -> +7/+11
- Air
  - [+] Damage: 1.0% -> 2.0%
  - [=] BKB: 47 -> 48
  - [=] KBG: 40 -> 30
  - [+] SDI Multiplier: 1.2x -> 1.0x

### Inhale:
- [+] FAF (eat, small/large): 38 -> 25

### Hammer Flip:
- Ground
  - [+] Charge Startup: 15 -> 6
  - [+] Hitbox Duration (uncharged): F21-22 -> F17-18
  - [+] FAF (release): 55 -> 47
- Air
  - (*) Fixed issue where the hitbox would persist when landing during the active frames
  - [-] Landing Lag: 12F -> 17F

### Final Cutter / Cutter Dash:
- Final Cutter
  - [-] Removed the ability to only perform the rising portion
- Cutter Dash
  - (!) Hold Special and either left or right before Kirby starts rising to transition into Cutter Dash
    - (!) Horizontal Cutter is actionable on hit in the air
  - Ground
    - Early
      - Hitbox Duration: F23-28
      - Damage: 10.0%
      - Angle: 72
      - BKB: 60
      - KBG: 48
      - Hitbox Size: 5.5u
    - Late
      - Hitbox Duration: F29-34
      - Damage: 10.0%
      - Angle: 72
      - BKB: 50
      - KBG: 48
      - Hitbox Size: 5.5u
      - FAF: 57
  - Air
    - Hitbox Duration: F23-36
    - Damage: 10.0%
    - Angle: 78
    - BKB: 80
    - KBG: 48
    - Hitbox Size: 5.0u
    - FAF (hit): 47

### Stone Transform:
- Air
  - [+] Startup (air): F29 -> F25
  - [+] Added light armor during frames 8-15
    - Threshold: 100KB
  - [-] Heavy armor threshold: 25.0% -> 15.0% (this is only during the startup)
- [+] FAF (un-transform): 32 -> 28

### Back Throw:
- [/] Angle: 130 -> 115

### Up Throw:
- [+] Damage: 10.0% -> 14.0%
- [/] KBG: 74 -> 68

### CODE REVIEW

File: fighters/kirby/src/acmd/tilts.rs [hunk 1]
```diff
@@ -8,9 +8,9 @@ unsafe fn game_attacks3hi(fighter: &mut L2CAgentBase) {
...
+        ATTACK(fighter, 1, 0, Hash40::new("footl"), 9.0, 60, 45, 0, 60, 3.3, 1.0, -4.5, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_F, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_KICK, *ATTACK_REGION_KICK);
+        ATTACK(fighter, 2, 0, Hash40::new("footl"), 9.0, 60, 45, 0, 60, 3.0, 1.0, -8.5, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_F, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_KICK, *ATTACK_REGION_KICK);
```
Comment (SuddyN) [inline fighters/kirby/src/acmd/tilts.rs:13]: you list the angle as 55, but here it's 60

File: fighters/kirby/src/status.rs [hunk 2]
```diff
@@ -1,12 +1,55 @@
...
+        *FIGHTER_STATUS_KIND_DAMAGE_FLY_ROLL,
+        *FIGHTER_STATUS_KIND_DAMAGE_FLY_METEOR,
+        *FIGHTER_STATUS_KIND_DAMAGE_FLY_REFLECT_LR,
+        *FIGHTER_STATUS_KIND_DAMAGE_FLY_REFLECT_U,
+        *FIGHTER_STATUS_KIND_DAMAGE_FLY_REFLECT_D,
+        *FIGHTER_STATUS_KIND_DAMAGE_FALL]) {
+        VarModule::off_flag(fighter.battle_object, vars::kirby::instance::DISABLE_SPECIAL_HI);
+    }
+    return true.into();
 }
```
Comment (SuddyN) [inline fighters/kirby/src/status.rs:53]: Are these actually necessary?

**Review by SuddyN — APPROVED** [review]

File: fighters/kirby/src/status.rs (hunk 2 above)
Comment (SkewedAskew) [reply fighters/kirby/src/status.rs:53]: Not anymore, good catch

Comment (SkewedAskew) [reply fighters/kirby/src/status.rs:53]: Wait no, I read that wrong. They are

**Pull request merged**